src/osemgrep/cli_login/Login_subcommand.ml

```diff
@@ -44,19 +44,130 @@ let parse_argv (argv : string array) : conf =
   let cmd : conf Cmd.t = Cmd.v cmdline_info cmdline_term in
   CLI_common.eval_value ~argv cmd
 
+let make_login_url () =
+  let session_id = Uuidm.v `V4 in
+  ( session_id,
+    Uri.(
+      add_query_params
+        (with_path Semgrep_envvars.env.semgrep_url "login")
+        [
+          ("cli-token", [ Uuidm.to_string session_id ]);
+          ( "docker",
+            [ (if Semgrep_envvars.env.in_docker then "True" else "False") ] );
+          ( "gha",
+            [ (if Semgrep_envvars.env.in_gh_action then "True" else "False") ]
+          );
+        ]) )
+
 (*****************************************************************************)
 (* Main logic *)
 (*****************************************************************************)
 
+let wait_between_retry_in_sec = 6 (* So every 10 retries is a minute *)
+let max_retries = 30 (* Give users 3 minutes to log in / open link *)
+
 (* All the business logic after command-line parsing. Return the desired
    exit code. *)
 let run (_conf : conf) : Exit_code.t =
-  (* TODO:
-     Setup_logging.setup config;
-     logger#info "Executed as: %s" (Sys.argv |> Array.to_list |> String.concat " ");
-     logger#info "Version: %s" config.version;
-  *)
-  Exit_code.ok
+  Logs_helpers.setup_logging ~force_color:false ~level:(Some Logs.Debug);
+  Logs.info (fun m -> m "Semgrep version: %s" Version.version);
```

Comment:
Should probably be Logs.debug.
Does semgrep login --verbose display those things?

Reply:
there's no `--verbose` for login/logout available.

Reply:
removed in [9306c04](https://github.com/returntocorp/semgrep/pull/7637/commits/9306c04c582e7d6724195fea1e0b102f0dc9335a)